lib/rosette/integrations/smartling-integration/tmx_document.rb

```diff
@@ -0,0 +1,72 @@
+# encoding: UTF-8
+
+require 'nokogiri'
+
+module Rosette
+  module Integrations
+    class SmartlingIntegration < Integration
+
+      # To be used when parsing .tmx files using Nokogiri. It turns out to be
+      # quite a bit faster to parse this way than parsing the whole document
+      # and looking for individual elements with xpath. We do sacrifice a bit
+      # of code readability, but the document really isn't that complicated.
+      class TmxDocument < Nokogiri::XML::SAX::Document
+        attr_reader :locale_code, :proc
+
+        def initialize(locale_code, &block)
+          @proc = block
+          @locale_code = locale_code
+          @capture_key = false
+          @capture_prop = false
+          @props = {}
+          @key = nil
+        end
+
+        def start_element(name, attrs = [])
+          case name
+            when 'tu'
```

Comment:
are these standard element names in a tmx file? If so, consider having them be constants